### DISCUSSION TITLE
the current iteration of the mod screen feels somewhat eclectic and doesn't use space as efficiently as it could

### DISCUSSION THREAD

**Discussion by tovc** [ppy/osu#29494] [Ideas]
Having individual mods' customization options be separate and distant from the mods themselves is confusing for new players, and isn't awfully utilitarian for the players that are actually used to it.
![image](https://github.com/user-attachments/assets/25bf25bf-fc68-427f-a612-bb92302dd8d2)

Additionally, on mobile and on larger UI scales, the search bar and customization menu take up an ungodly amount of space, and make using the mod screen a chore. I believe this was updated recently to make individual mods smaller, but the mod screen still barely shows 4 mods, and more than half of the vertical space is taken up by unrelated title bars, scrollbars and the search bar/customize dropdown.
![Screenshot_2024-08-18-22-16-11-920_sh ppy osulazer](https://github.com/user-attachments/assets/7c06e5f9-4bd3-4882-8816-9f7d27c4996e)

It's my understanding and hope that the customization menu is a somewhat temporary solution until the song selection carousel redesign is implemented. In it's current state, the 'customization' dropdown and the search bar feel foreign to the rest of the screen. The search bar is a parallelogram, as if it's trying to fit in with the rest of the screen's design cues, but the customize dropdown is square, and just doesn't fit in. I believe that moving the search bar to the mod screen title bar was mentioned a while back, but it hasn't happened yet, so I'm here to suggest it again.

My ideal mod screen redesign would look something like this:
![osu_2024-08-18_20-53-32](https://github.com/user-attachments/assets/ec35bcb2-9e9a-495f-9abb-ef1a2e849382)

Move the individual mod settings to a dropdown below the mods themselves. This is a lot more intuitive for new players, as well as entirely removing the need for the separate 'customization' dropdown. Move the search bar to the title bar and remove the little 'X' at the right side. This frees up even more vital vertical space allowing for more mods, especially on mobile.

There is certainly more that could be improved, and more space that could be saved, but I think this would go a long way to making the screen more coherent, clean and intuitive.

**Comment by Gabixel** [comment]
Probably worth mentioning [this previous answer from the devs](https://github.com/ppy/osu/discussions/18507#discussioncomment-2855646) to my post with a similar design suggestion (for the customization part). In short, I think they prefer the current dropdown design and believe that it looks bad with multiple settings for a single mod.

**Reply by tovc** [reply]
my 'design suggestion' is somewhat just put forward as a way to avoid exclusively putting forward criticisms of the current screen. I do think that having multiple options beneath the mods would still work better than the current system. I've watched countless stable players try lazer, and they nearly universally get stumped by the mod customization screen. I have no doubt that there's a better way to do this than what I've put forward, but I think it's better than the current layout
